src/main/webapp/dev/js/main/adminHome.es6

```diff
@@ -56,191 +195,136 @@ function disableAddInstructorForm() {
 }
 
 /**
- * Sends Ajax request to add new instructor(s).
- * It only sends another Ajax request after it finishes.
+ * Is called as part of the callback to the AJAX call used to add instructors.
+ * Updates the table that displays whether adding the instructor was successful.
+ * In the event of an error, preserves the text entered into the forms so that a
+ * retry can be attempted.
  */
-const addInstructorByAjaxRecursivelyImpl = () => {
-    $.ajax({
-        type: 'POST',
-        url: `/admin/adminInstructorAccountAdd?${makeCsrfTokenParam()}&${paramsList[paramsCounter]}`,
-        beforeSend: disableAddInstructorForm,
-        error() {
-            addInstructorAjax(true, null);
-        },
-        success(data) {
-            addInstructorAjax(false, data);
-        },
-    });
-};
+function updateInstructorAddStatus(ajaxResults) {
+    for (let i = 0; i < ajaxResults.length; i += 1) {
+        const ajaxResult = ajaxResults[i];
 
-let addInstructorByAjaxRecursively = addInstructorByAjaxRecursivelyImpl;
+        const shortName = ajaxResult.isError() ? '-' : ajaxResult.responseData.instructorShortName;
+        const name = ajaxResult.isError() ? '-' : ajaxResult.responseData.instructorName;
+        const email = ajaxResult.isError() ? '-' : ajaxResult.responseData.instructorEmail;
+        const institution = ajaxResult.isError() ? '-' : ajaxResult.responseData.instructorInstitution;
+        const isSuccess = !ajaxResult.isError() && !ajaxResult.isAddFailed();
+        const status = ajaxResult.getStatusMessage();
 
-/**
- * Reads information of instructor(s) from the first panel and adds each instructor one by one.
- */
-function addInstructorFromFirstFormByAjax() {
-    $('#addInstructorResultPanel').show();    // show the hidden panel
-    isInputFromFirstPanel = true;
-
-    let multipleLineText = $('#addInstructorDetailsSingleLine').val();    // get input from the first panel
-    multipleLineText = multipleLineText.trim();
-    if (multipleLineText.length > 0) {
-        instructorDetailsList = multipleLineText.split('\n');
-        paramsList = [];
-        for (let i = 0; i < instructorDetailsList.length; i += 1) {
-            instructorDetailsList[i] = instructorDetailsList[i].replace(/\t/g, '|');
-            paramsList[i] = `instructordetailssingleline=${encodeURIComponent(instructorDetailsList[i])}`;
-        }
-    }
-    paramsCounter = 0;
-    $('#addInstructorResultTable tbody').html('');    // clear table
-    $('#addInstructorDetailsSingleLine').val('');    // clear input form
-    $('#addInstructorResultPanel div.panel-heading').html('<strong>Result</strong>');    // clear panel header
-    if (paramsList.length > 0) {
-        addInstructorByAjaxRecursively();
+        const rowText = createRowForResultTable(shortName, name, email, institution, isSuccess, status);
+        $('#addInstructorResultTable tbody').append(rowText);
+        const panelHeader = `<strong>Result (${i + 1}/${ajaxResults.length})</strong>`;
```

Comment:
Otherwise the admin will have to wait very long when adding multiple instructors, with little indication of progress.